go/private/rules/cgo.bzl

```diff
@@ -57,6 +57,40 @@ load(
 
 _CgoCodegen = provider()
 _CgoInfo = provider()
+_CcAlwaysLinkInfo = provider()
+
+_WHOLE_ARCHIVE_FLAGS_BY_GOOS = {
```

Comment:
I don't think the `{} + {}` operator is supported anymore in Starlark. If it is, it's deprecated.

I think it would be simpler to just define a function that returns the args though.

Reply:
I just tested on Bazel 0.21 and it worked without warning. Where did you see that?

Reply:
I don't think it will work with --all_incompatible_changes.

See https://docs.bazel.build/versions/master/skylark/backward-compatibility.html#dictionary-concatenation